WordPress/openverse-frontend
# Rename media store getters

## Fixes


Fixes #830 by @obulat

## Description


This PR renames some of the `media` store `getters` so that there are no `getters` with the same name as a state property. This is necessary because both `getters` and state properties will be available as properties of `this` in Pinia.

While renaming, I did a couple of simplifications, too:
- all media items get assigned `frontendMediaType` property by the media service, so I removed the addition  of `frontendMediaType` from the `VAllResultsGrid` component.
- `results` getter was only used by the `VContentLink` component to access the result counts for each media type, so I rewrote the getter as `mediaResultCounts` that only returns `{ mediaType: N }` instead of passing all of the data.
- reverted the conversion of `items` from an Array to an Object. We only look up an item by its id in the Global Audio Player. Everywhere else we use items as an array, and because of that had to convert the items object into an array in at least 3-4 places.
- 
## Testing Instructions

Run the app, try selecting different search types, clicking the 'Load more items' button, select various filters. Also, try to load each kind of pages from the server (i.e. go after searching for something or opening a single result page, refresh the page). There should be no errors or warnings.

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
